src/buffers/disk/leveldb_buffer.rs

```diff
@@ -320,7 +327,10 @@ impl Reader {
             self.uncompacted_size = 0;
 
             debug!("Compacting disk buffer.");
-            self.db.compact(&Key(0), &Key(self.delete_offset));
+            self.db
+                .compact(&Key(self.uncompacted_offset), &Key(self.delete_offset));
+
+            self.uncompacted_offset = self.delete_offset;
```

Comment:
Does this also need atomics? Or are we the only reader/writer? I'm just a bit confused why some of the fields in `Reader` are atomic and others are not. Some comments could help.

Reply:
Some of the variables are shared between this only reader and N writers so they require atomics but this variable is self contained in the reader. Yep, some comments would, I'll add them.

Reply:
Thanks 😄 They'd definitely help me follow it better.

Reply:
@jszwedko check the new comments. We can add more if you think more is needed.

Reply:
Those are perfect, thanks @ktff !